components/telemetry-controller/controllers/loggingconfiguration_controller.go

```diff
@@ -0,0 +1,318 @@
+		log.Error(err, "failed to sync ConfigMap")
+		return ctrl.Result{}, err
+	}
+
+	updatedFiles, err := r.syncFiles(ctx, &config)
+	if err != nil {
+		log.Error(err, "failed to sync mounted files")
+		return ctrl.Result{}, err
+	}
+
+	updatedEnv, err := r.syncEnvironment(ctx, &config)
+	if err != nil {
+		log.Error(err, "failed to sync environment variables")
+		return ctrl.Result{}, err
+	}
+
+	if updatedCm || updatedFiles || updatedEnv {
+		if err := r.Update(ctx, &config); err != nil {
+			log.Error(err, "cannot update LoggingConfiguration")
+			return ctrl.Result{}, err
+		}
+
+		if err := r.deleteFluentBitPods(ctx, log); err != nil {
+			log.Error(err, "cannot delete fluent bit pods")
+			return ctrl.Result{}, err
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *LoggingConfigurationReconciler) getOrCreateConfigMap(ctx context.Context, name types.NamespacedName) (corev1.ConfigMap, error) {
+	var cm corev1.ConfigMap
+	if err := r.Get(ctx, name, &cm); err != nil {
+		if errors.IsNotFound(err) {
+			cm = corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name.Name,
+					Namespace: name.Namespace,
+				},
+			}
+			if err := r.Create(ctx, &cm); err != nil {
+				return cm, err
+			}
+		} else {
+			return cm, err
+		}
+	}
+	return cm, nil
+}
+
+func (r *LoggingConfigurationReconciler) syncConfigMap(ctx context.Context, config *telemetryv1alpha1.LoggingConfiguration) (bool, error) {
+	cm, err := r.getOrCreateConfigMap(ctx, r.FluentBitConfigMap)
+	if err != nil {
+		return false, err
+	}
+	cmKey := config.Name + ".conf"
+
+	// Add or remove Fluent Bit configuration sections
+	if config.DeletionTimestamp != nil {
+		if cm.Data != nil && controllerutil.ContainsFinalizer(config, configMapFinalizer) {
+			delete(cm.Data, cmKey)
+			controllerutil.RemoveFinalizer(config, configMapFinalizer)
+		}
+	} else {
+		fluentBitConfig := generateFluentBitConfig(config.Spec.Sections)
+		if cm.Data == nil {
+			data := make(map[string]string)
+			data[cmKey] = fluentBitConfig
+			cm.Data = data
+		} else {
+			if oldConfig, hasKey := cm.Data[cmKey]; hasKey && oldConfig == fluentBitConfig {
+				// Nothing changed
+				return false, nil
+			}
+			cm.Data[cmKey] = fluentBitConfig
+		}
+		controllerutil.AddFinalizer(config, configMapFinalizer)
+	}
+
+	// Update ConfigMap
+	if err := r.Update(ctx, &cm); err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func (r *LoggingConfigurationReconciler) syncFiles(ctx context.Context, config *telemetryv1alpha1.LoggingConfiguration) (bool, error) {
+	changed := false
+	cm, err := r.getOrCreateConfigMap(ctx, r.FluentBitFilesConfigMap)
+	if err != nil {
+		return false, err
+	}
+
+	// Sync files from every section
+	for _, section := range config.Spec.Sections {
+		for _, file := range section.Files {
+			if config.DeletionTimestamp != nil {
+				if _, hasKey := cm.Data[file.Name]; hasKey {
+					delete(cm.Data, file.Name)
+					controllerutil.RemoveFinalizer(config, filesFinalizer)
```

Comment:
Should we check if it contains a finalizer similar to what we do above: https://github.com/kyma-project/kyma/pull/11720/files#diff-d637cd02f4ed5dd941b75c86db5467f035f3856c5945365e5299bce15b710bc2R142?

Reply:
This check is already done in the `RemoveFinalizer` method